cmd/podman/common/createparse.go

```diff
@@ -9,7 +9,7 @@ import (
 // by validate must not need any state information on the flag (i.e. changed)
 func (c *ContainerCLIOpts) validate() error {
 	var ()
-	if c.Rm && c.Restart != "" && c.Restart != "no" {
+	if c.Rm && (c.Restart != "" && c.Restart != "no" && c.Restart != "on-failure") {
```

Comment:
It would be nice to have "on-failure" defined in a variable somewhere rather than a bare string scattered about.  Not for this pr though.

Reply:
It is defined in the `libpod` package as `RestartPolicyOnFailure`.
The same for `no` it should be `RestartPolicyNo`.
https://github.com/containers/podman/blob/dc58d4e2858bf6688a775277adf8f775afc30e73/libpod/container.go#L99

Reply:
@rhatdan are you going to change per @Luap99 's suggestion?